extensions/notebook/src/jupyter/jupyterServerInstallation.ts

```diff
@@ -708,6 +711,43 @@ export class JupyterServerInstallation implements IJupyterServerInstallation {
 	public getRequiredPackagesForKernel(kernelName: string): PythonPkgDetails[] {
 		return this._requiredKernelPackages.get(kernelName) ?? [];
 	}
+
+	public get runningOnSaw(): boolean {
+		return this._runningOnSAW;
+	}
+
+	public async updateKernelSpecPaths(kernelsFolder: string): Promise<void> {
+		if (!this._runningOnSAW || this._kernelSpecsUpdated) {
+			return;
+		}
+
+		let fileNames = await fs.readdir(kernelsFolder);
+		let filePaths = fileNames.map(name => path.join(kernelsFolder, name));
+		let fileStats = await Promise.all(filePaths.map(path => fs.stat(path)));
+		let folderPaths = filePaths.filter((value, index) => value && fileStats[index].isDirectory());
+
+		let kernelFiles = folderPaths.map(folder => path.join(folder, 'kernel.json'));
+		if (this._runningOnSAW) {
```

Comment:
This if-else case shouldn't be necessary. We already retrieve the kernel file paths in the code directly above this.

Reply:
Or you could remove that general kernel searching code and just have a list of specific kernel files like you're doing in this if-else. In that case we'd only be adding powershell to the list of kernel files.

Reply:
We shouldn't be referencing the python3 kernelspec at all.

Reply:
makes sense. removed the if-else.